src/leaspy/io/data/data.py

```
@@ -529,25 +541,38 @@ def from_individual_values(
             if not event_bool_name or not event_time or not event_bool:
                 raise ("Not coherent inputs for longitudinal data")
 
+        # Covariates input checks
+        if not covariate_names:
```

Small trick to simplify these conditions.
Also, please raise a proper error with a message as helpful as possible for the user:

```python
if (covariate_names is None) == (covariates is None):
    raise ValueError(
        "Not coherent inputs for covariate data: \n "
        f"covariate_names = {covariate_names} and \n "
        f"covariates = {covariates}."
    )
```

---

When in doubt, it's always useful to open an the interpreter and try:

```python
>>> def f(x, y):
...     if (x is None) == (y is None):
...         print (f"x = {x}, y = {y}")
...     else:
...         raise ValueError("bad bad bad")
...
>>> f(1, 2)
x = 1, y = 2
>>> f(None, None)
x = None, y = None
>>> f(1, None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in f
ValueError: bad bad bad
>>> f(None, 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in f
ValueError: bad bad bad
```

Is that the desired behavior ?

---

Thanks for the tip. I only want an `if`condition (not `if` `else`) so I prefer this form :
```python
if (covariate_names is None) != (covariates is None):
    raise ValueError("Not coherent inputs for covariate data: \n "
        f"covariate_names = {covariate_names} and \n "
        f"covariates = {covariates}.")
```
So that the error is raised when one is `None`and the other is not.

---

I misunderstood your question, sorry about that.
Yes, you should totally do that 👍